src/validator/validationManager.ts

```diff
@@ -374,4 +417,83 @@ export class ValidationManager {
     const isPolygon = compiledSchema(footprint);
     return isPolygon;
   }
+
+  //#region validate sources
+
+  @withSpanV4
+  private async isValidTilesetContent(fullPath: string, isValidTilesetContentResponse: IsValidTilesetContentResponse): Promise<boolean> {
+    const logContext = { ...this.logContext, function: this.isValidTilesetContent.name };
+    try {
+      this.logger.debug({
+        msg: 'Tileset validation started',
+        logContext,
+        fullPath: fullPath,
+      });
+      const fileContent: string = await readFile(fullPath, { encoding: FILE_ENCODING });
+      const tileSetJson: TileSetJson = JSON.parse(fileContent) as TileSetJson;
+      const shape = tileSetJson.root.boundingVolume;
+
+      if (shape.sphere != undefined) {
+        isValidTilesetContentResponse.polygon = convertSphereFromXYZToWGS84(shape as BoundingSphere);
+      } else if (shape.region != undefined) {
+        isValidTilesetContentResponse.polygon = convertRegionFromRadianToDegrees(shape as BoundingRegion);
+      } else if (shape.box != undefined) {
+        isValidTilesetContentResponse.message = `BoundingVolume of box is not supported yet... Please contact 3D team.`;
+        return false;
+      } else {
+        isValidTilesetContentResponse.message = 'Bad tileset format. Should be in 3DTiles format';
+        return false;
+      }
+
+      // TODO: refactor in next PR in order to return bool and better message
+      const validatePolygonResult = this.validatePolygon(isValidTilesetContentResponse.polygon);
+      if (typeof validatePolygonResult == 'string') {
+        isValidTilesetContentResponse.message = validatePolygonResult;
+        return false;
+      }
+    } catch (error) {
+      const message = `File '${fullPath}' tileset validation failed`;
+      this.logger.error({
+        msg: message,
+        logContext,
+        fullPath,
+        error,
+      });
+      isValidTilesetContentResponse.message = message;
+      return false;
+    }
+    return true;
+  }
+
+  @withSpanV4
+  private async validateFilesExist(fullPath: string): Promise<boolean> {
```

Comment:
You wrote a new function that simply execute one fs operation. Why not just call the operation without set it in a new function?

Reply:
2 reasons:
* because i wan't the logs (before and after) with same paramaters
* The function contains 10 rows that will make the caller code "ugly";

Reply:
OK